agent/structs/structs.go

```diff
@@ -629,6 +640,30 @@ func (s *NodeService) IsSame(other *NodeService) bool {
 	return true
 }
 
+// IsSame checks if one ServiceNode is the same as another, without looking
+// at the Raft information (that's why we didn't call it IsEqual). This is
+// useful for seeing if an update would be idempotent for all the functional
+// parts of the structure.
+func (s *ServiceNode) IsSame(other *ServiceNode) bool {
```

Comment:
Is there a reason this doesn't check TaggedAddresses, NodeMeta and Datacenter?

Is that because during registration your IsSame check on the node will already handle checking all of that?

Reply:
Yes, I assumed the node will be updated in that case, but thinking about it, this is probably a false assumption for another user of this new function.

It probably does not change anything, but I added those entities in next patch, so another developer won't be bitten by my current assumptions.